tests/components/withings/test_init.py

```diff
@@ -275,9 +314,211 @@ async def test_config_flow_upgrade(
+        assert hass.config_entries.async_entries(DOMAIN)
+
+        for config_entry in hass.config_entries.async_entries("withings"):
+            await hass.config_entries.async_remove(config_entry.entry_id)
+            fake_delete_cloudhook.assert_called_once()
+
+        await hass.async_block_till_done()
+        assert not hass.config_entries.async_entries(DOMAIN)
+
+
+async def test_setup_without_https(
+    hass: HomeAssistant,
+    webhook_config_entry: MockConfigEntry,
+    withings: AsyncMock,
+    caplog: pytest.LogCaptureFixture,
+) -> None:
+    """Test if set up with cloud link and without https."""
+    hass.config.components.add("cloud")
+    with patch(
+        "homeassistant.helpers.network.get_url",
+        return_value="http://example.nabu.casa",
+    ), patch(
+        "homeassistant.helpers.config_entry_oauth2_flow.async_get_config_entry_implementation",
+    ), patch(
+        "homeassistant.components.withings.webhook_generate_url"
+    ) as mock_async_generate_url:
+        mock_async_generate_url.return_value = "http://example.com"
+        await setup_integration(hass, webhook_config_entry)
+
+        await hass.async_block_till_done()
+        mock_async_generate_url.assert_called_once()
+
+    assert "https and port 443 is required to register the webhook" in caplog.text
+
+
+async def test_cloud_disconnect(
+    hass: HomeAssistant,
+    withings: AsyncMock,
+    webhook_config_entry: MockConfigEntry,
+    hass_client_no_auth: ClientSessionGenerator,
+    freezer: FrozenDateTimeFactory,
+) -> None:
+    """Test disconnecting from the cloud."""
+    await mock_cloud(hass)
+    await hass.async_block_till_done()
+
+    with patch(
+        "homeassistant.components.cloud.async_is_logged_in", return_value=True
+    ), patch(
+        "homeassistant.components.cloud.async_is_connected", return_value=True
+    ), patch(
+        "homeassistant.components.cloud.async_active_subscription", return_value=True
+    ), patch(
+        "homeassistant.components.cloud.async_create_cloudhook",
+        return_value="https://hooks.nabu.casa/ABCD",
+    ), patch(
+        "homeassistant.helpers.config_entry_oauth2_flow.async_get_config_entry_implementation",
+    ), patch(
+        "homeassistant.components.cloud.async_delete_cloudhook"
+    ), patch(
+        "homeassistant.components.withings.webhook_generate_url"
+    ):
+        await setup_integration(hass, webhook_config_entry)
+        assert hass.components.cloud.async_active_subscription() is True
+        assert hass.components.cloud.async_is_connected() is True
+
+        await hass.async_block_till_done()
+
+        withings.async_notify_list.return_value = NotifyListResponse(
+            **load_json_object_fixture("withings/empty_notify_list.json")
+        )
+
+        assert withings.async_notify_subscribe.call_count == 6
+
+        async_dispatcher_send(
+            hass, SIGNAL_CLOUD_CONNECTION_STATE, CloudConnectionState.CLOUD_DISCONNECTED
```

Comment:
We shouldn't use this signal directly. It should be triggered from the cloud integration somehow.

Reply:
Do you have any tips on how to trigger it?

Reply:
I think we could make a common test helper that uses the dispatcher signal. That way we don't need to mock out the cloud integration in the tests of other integrations and we don't use cloud integration details in the tests of other integrations.

Reply:
So we still use this same piece of code but we encapsulate it in a helper?

Reply:
Yes.

Reply:
What would the best place for this helper be?

Reply:
Probably `tests/common.py`.